server/resource/caldav_service.go

```diff
@@ -1,18 +1,545 @@
 package resource
 
 import (
+	"database/sql"
+	"encoding/base64"
+	"fmt"
+	"github.com/daptin/daptin/server/database"
+	"github.com/jmoiron/sqlx"
 	"github.com/samedi/caldav-go"
+	"github.com/samedi/caldav-go/data"
+	"github.com/samedi/caldav-go/errs"
+	log "github.com/sirupsen/logrus"
 	"net/http"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
 )
 
-func NewCaldavServer(CaldavListenInterface string)*http.Server{
+var Schema = `
+CREATE EXTENSION IF NOT EXISTS pgcrypto;
```

Comment:
this postgres specific, we need to work with mysql/postgres/sqlite. the "encrypted" column type will encrypt the values before storing, check the oauth_token table for example, https://github.com/daptin/daptin/blob/master/server/resource/columns.go#L2422